### ISSUE TITLE
Dimension Mismatch error in [image_captioning.ipynb]

### ISSUE THREAD

**Issue by muneeb699** [tensorflow/tensorflow#27862]
<em>Please make sure that this is a documentation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:doc_template</em>

**System information**
- TensorFlow version:2.2.0 alpha
- Doc Link: [https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/r2/tutorials/sequences/image_captioning.ipynb](url)

I am running this example with the Google colab and getting this issue: 
```
InvalidArgumentError                      Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs)
   1818   try:
-> 1819     c_op = c_api.TF_FinishOperation(op_desc)
   1820   except errors.InvalidArgumentError as e:

InvalidArgumentError: Dimension 0 in both shapes must be equal, but are 1 and 32. Shapes are [1,1] and [32,1]. for 'rnn__decoder/concat' (op: 'ConcatV2') with input shapes: [1,1,256], [32,1,256], [] and with computed input tensors: input[2] = <-1>.

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-40-bdc59db794fd> in <module>()
      6 
      7     for (batch, (img_tensor, target)) in enumerate(dataset):
----> 8         batch_loss, t_loss = train_step(img_tensor, target)
      9         total_loss += t_loss
     10 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py in __call__(self, *args, **kwds)
    412       # In this case we have created variables on the first call, so we run the
    413       # defunned version which is guaranteed to never create variables.
--> 414       return self._stateless_fn(*args, **kwds)  # pylint: disable=not-callable
    415     elif self._stateful_fn is not None:
    416       # In this case we have not created variables on the first call. So we can

/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py in __call__(self, *args, **kwargs)
   1285   def __call__(self, *args, **kwargs):
   1286     """Calls a graph function specialized to the inputs."""
-> 1287     graph_function, args, kwargs = self._maybe_define_function(args, kwargs)
   1288     return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
   1289 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py in _maybe_define_function(self, args, kwargs)
   1609           relaxed_arg_shapes)
   1610       graph_function = self._create_graph_function(
-> 1611           args, kwargs, override_flat_arg_shapes=relaxed_arg_shapes)
   1612       self._function_cache.arg_relaxed[rank_only_cache_key] = graph_function
   1613 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py in _create_graph_function(self, args, kwargs, override_flat_arg_shapes)
   1510             arg_names=arg_names,
   1511             override_flat_arg_shapes=override_flat_arg_shapes,
-> 1512             capture_by_value=self._capture_by_value),
   1513         self._function_attributes)
   1514 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/func_graph.py in func_graph_from_py_func(name, python_func, args, kwargs, signature, func_graph, autograph, autograph_options, add_control_dependencies, arg_names, op_return_value, collections, capture_by_value, override_flat_arg_shapes)
    692                                           converted_func)
    693 
--> 694       func_outputs = python_func(*func_args, **func_kwargs)
    695 
    696       # invariant: `func_outputs` contains only Tensors, IndexedSlices,

/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py in wrapped_fn(*args, **kwds)
    315         # __wrapped__ allows AutoGraph to swap in a converted function. We give
    316         # the function a weak reference to itself to avoid a reference cycle.
--> 317         return weak_wrapped_fn().__wrapped__(*args, **kwds)
    318     weak_wrapped_fn = weakref.ref(wrapped_fn)
    319 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/func_graph.py in wrapper(*args, **kwargs)
    684                   optional_features=autograph_options,
    685                   force_conversion=True,
--> 686               ), args, kwargs)
    687 
    688         # Wrapping around a decorator allows checks like tf_inspect.getargspec

/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py in converted_call(f, owner, options, args, kwargs)
    390     return _call_unconverted(f, args, kwargs)
    391 
--> 392   result = converted_f(*effective_args, **kwargs)
    393 
    394   # The converted function's closure is simply inserted into the function's

/tmp/tmp3uq18dwt.py in tf__train_step(img_tensor, target)
     15       dec_input_1 = ag__.converted_call('expand_dims', tf, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_6, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (target[:, (i)], 1), {})
     16       return dec_input_1, loss_1, hidden_1
---> 17     dec_input, loss, hidden = ag__.for_stmt(ag__.converted_call(range, None, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_3, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (1, target.shape[1]), {}), None, loop_body, (dec_input, loss, hidden))
     18   total_loss = loss / ag__.converted_call(int, None, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_7, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (target.shape[1],), {})
     19   trainable_variables = encoder.trainable_variables + decoder.trainable_variables

/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/operators/control_flow.py in for_stmt(iter_, extra_test, body, init_state)
     79     return _dataset_for_stmt(iter_, extra_test, body, init_state)
     80   else:
---> 81     return _py_for_stmt(iter_, extra_test, body, init_state)
     82 
     83 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/operators/control_flow.py in _py_for_stmt(iter_, extra_test, body, init_state)
     88     if extra_test is not None and not extra_test(*state):
     89       break
---> 90     state = body(target, *state)
     91   return state
     92 

/tmp/tmp3uq18dwt.py in loop_body(loop_vars, dec_input_1, loss_1, hidden_1)
     11     def loop_body(loop_vars, dec_input_1, loss_1, hidden_1):
     12       i = loop_vars
---> 13       predictions, hidden_1, _ = ag__.converted_call(decoder, None, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_4, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (dec_input_1, features, hidden_1), {})
     14       loss_1 += ag__.converted_call(loss_function, None, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_5, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (target[:, (i)], predictions), {})
     15       dec_input_1 = ag__.converted_call('expand_dims', tf, ag__.ConversionOptions(recursive=True, verbose=0, strip_decorators=(tf.function, defun_6, ag__.convert, ag__.do_not_convert, ag__.converted_call), force_conversion=False, optional_features=(), internal_convert_user_code=True), (target[:, (i)], 1), {})

/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py in converted_call(f, owner, options, args, kwargs)
    265 
    266   if not options.force_conversion and conversion.is_whitelisted_for_graph(f):
--> 267     return _call_unconverted(f, args, kwargs)
    268 
    269   # internal_convert_user_code is for example turned off when issuing a dynamic

/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py in _call_unconverted(f, args, kwargs)
    186     return f.__self__.call(args, kwargs)
    187 
--> 188   return f(*args, **kwargs)
    189 
    190 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, inputs, *args, **kwargs)
    610                       base_layer_utils.AutoAddUpdates(self,
    611                                                       inputs)) as auto_updater:
--> 612                 outputs = self.call(inputs, *args, **kwargs)
    613                 auto_updater.set_outputs(outputs)
    614 

<ipython-input-33-dc6f8641457a> in call(self, x, features, hidden)
     22 
     23     # x shape after concatenation == (batch_size, 1, embedding_dim + hidden_size)
---> 24     x = tf.concat([tf.expand_dims(context_vector, 1), x], axis=-1)
     25 
     26     # passing the concatenated vector to the GRU

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/dispatch.py in wrapper(*args, **kwargs)
    178     """Call target, and fall back on dispatchers if there is a TypeError."""
    179     try:
--> 180       return target(*args, **kwargs)
    181     except (TypeError, ValueError):
    182       # Note: convert_to_eager_tensor currently raises a ValueError, not a

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/array_ops.py in concat(values, axis, name)
   1269               tensor_shape.scalar())
   1270       return identity(values[0], name=scope)
-> 1271   return gen_array_ops.concat_v2(values=values, axis=axis, name=name)
   1272 
   1273 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_array_ops.py in concat_v2(values, axis, name)
   1215   _attr_N = len(values)
   1216   _, _, _op = _op_def_lib._apply_op_helper(
-> 1217         "ConcatV2", values=values, axis=axis, name=name)
   1218   _result = _op.outputs[:]
   1219   _inputs_flat = _op.inputs

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py in _apply_op_helper(self, op_type_name, name, **keywords)
    798         op = g.create_op(op_type_name, inputs, output_types, name=scope,
    799                          input_types=input_types, attrs=attr_protos,
--> 800                          op_def=op_def)
    801       return output_structure, op_def.is_stateful, op
    802 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/func_graph.py in create_op(self, op_type, inputs, dtypes, input_types, name, attrs, op_def, compute_shapes, compute_device)
    450     return super(FuncGraph, self).create_op(
    451         op_type, inputs, dtypes, input_types, name, attrs, op_def,
--> 452         compute_device=compute_device)
    453 
    454   def capture(self, tensor, name=None):

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    505                 'in a future version' if date is None else ('after %s' % date),
    506                 instructions)
--> 507       return func(*args, **kwargs)
    508 
    509     doc = _add_deprecated_arg_notice_to_docstring(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in create_op(***failed resolving arguments***)
   3477           input_types=input_types,
   3478           original_op=self._default_original_op,
-> 3479           op_def=op_def)
   3480       self._create_op_helper(ret, compute_device=compute_device)
   3481     return ret

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in __init__(self, node_def, g, inputs, output_types, control_inputs, input_types, original_op, op_def)
   1981           op_def, inputs, node_def.attr)
   1982       self._c_op = _create_c_op(self._graph, node_def, grouped_inputs,
-> 1983                                 control_input_ops)
   1984 
   1985     # Initialize self._outputs.

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs)
   1820   except errors.InvalidArgumentError as e:
   1821     # Convert to ValueError for backwards compatibility.
-> 1822     raise ValueError(str(e))
   1823 
   1824   return c_op

ValueError: Dimension 0 in both shapes must be equal, but are 1 and 32. Shapes are [1,1] and [32,1]. for 'rnn__decoder/concat' (op: 'ConcatV2') with input shapes: [1,1,256], [32,1,256], [] and with computed input tensors: input[2] = <-1>.
```

**I checked different thing but can't able to solve this issue**

**We welcome contributions by users. Will you be able to update submit a PR (use the [doc style guide](https://www.tensorflow.org/community/documentation)) to fix the doc Issue?**

**Comment by monjoybme** [comment]
I am unable to access your` image_captioning.ipynb`. Hence, it is difficult to rectify your issue without seeing your code. From your error message I can suggest you that check dimensions of each layer. You did mistake in defining input shapes and corresponding output shapes. Concentrate in the below error lines. 

`InvalidArgumentError: Dimension 0 in both shapes must be equal, but are 1 and 32. Shapes are [1,1] and [32,1]. for 'rnn__decoder/concat' (op: 'ConcatV2') with input shapes: [1,1,256], [32,1,256], [] and with computed input tensors: input[2] = <-1>.`

**Comment by gadagashwini** [comment]
@muneeb699 Please provide the correct google colab link to reproduce the issue reported here. Thanks!

**Comment by muneeb699** [comment]
Hi Here is the link :
[https://colab.research.google.com/drive/1lORosF4OMpsG2iEhyJzAXikEQEcViWE2](url)

**Comment by gadagashwini** [comment]
@muneeb699 Still the colab link that you have provided is not correct. Please provide correct link. Thanks!

**Comment by muneeb699** [comment]
Copy the link and paste it. It will not be open when you click it. 
[https://colab.research.google.com/drive/1lORosF4OMpsG2iEhyJzAXikEQEcViWE2](url)

**Comment by monjoybme** [comment]
Your problem has been solved. Change **BATCH_SIZE = 48**

# feel free to change these parameters according to your system's configuration

```
BATCH_SIZE = 48
BUFFER_SIZE = 1000
embedding_dim = 256
units = 512
vocab_size = len(tokenizer.word_index) + 1
num_steps = len(img_name_train) // BATCH_SIZE
# shape of the vector extracted from InceptionV3 is (64, 2048)
# these two variables represent that
features_shape = 2048
attention_features_shape = 64
```

**Comment by muneeb699** [comment]
Why this problem is occured in the first place? Because you just change the batch size and it gives an error. Doesn't seems right

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow-tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

**Comment by MarkDaoust** [comment]
The batch size missmatch was fixed in tensorflow/docs@9e7ea7c446e7f40b3d1
